widget/bind_helper_test.go

```diff
@@ -1,43 +1,49 @@
 package widget
 
 import (
+	"sync/atomic"
 	"testing"
 
 	"fyne.io/fyne/v2/data/binding"
 	"github.com/stretchr/testify/assert"
 )
 
+type atomicInt struct{ val int64 }
+
+func (v *atomicInt) Set(val int) { atomic.StoreInt64(&v.val, int64(val)) }
+func (v *atomicInt) Get() int    { return int(atomic.LoadInt64(&v.val)) }
```

Comment:
using `atomic.LoadInt64` will crash on 32bit as we saw.
Is it your intention to limit the unit tests to 64bit machines?

Reply:
I think we won't have a chance to run tests on 32bit machine, would we?

Reply:
Apart from on macOS, you should be able to compile and run 32bit binaries on a 64-but machine. We could try running them in our tests as well.

Reply:
I was also thinking that it means someone on a 32bit machine looking to compile would be restricted from getting test passing. It does not seem desirable. 
apart from that I suppose it means we can’t use a quick grep any more to make sure we are not using LoadInt64!

Reply:
I feel horrible about thinking in a way to prevent people from using 64bit functions.
Furthermore, I could not even pass all tests on my 64bit machine today but now an argument about a single test may not pass on a 32bit machine is not desirable!

The sync/atomic doc only says:

- On 386, the 64-bit functions use instructions unavailable before the Pentium MMX.
- On non-Linux ARM, the 64-bit functions use instructions unavailable before the ARMv6k core.
- On ARM, 386, and 32-bit MIPS, it is the caller's responsibility to arrange for 64-bit alignment of 64-bit words accessed atomically. The first word in a variable or in an allocated struct, array, or slice can be relied upon to be 64-bit aligned.

https://pkg.go.dev/sync/atomic@go1.17.2#pkg-note-BUG

This means:

1. Only ARMs before  Pentium MMX and ARMv6k is not usable
2. If the developer properly handles the struct alignment, 64-bit atomics can still work properly on ARM, 386, and 32-bit MIPS.

BTW: The runtime garbage collector is using [atomic.Loadint64](https://cs.opensource.google/go/go/+/master:src/runtime/mgcmark.go;l=418?q=LoadInt64&ss=go%2Fgo&start=21), does that mean we can't even use Go at all??

Reply:
They should work, you "just" need to align them correctly. I have seen people either using a [2]uint64 or putting the uint64 at the start of the struct fields.

Reply:
> BTW: The runtime garbage collector is using atomic.Loadint64, does that mean we can't even use Go at all??

I have no idea expertise in this area - I am just referencing the bug we worked on before that crashed apps on 32bit devices.
I would assume that the GC has some clever padding to avoid the issue, as clearly Go is working fine ok 32bit